auraed/src/runtime/cells/cell_service.rs

```diff
@@ -98,44 +102,100 @@ impl CellService {
     async fn start(
         &self,
         request: ValidatedStartExecutableRequest,
-    ) -> Result<StartExecutableResponse> {
-        let ValidatedStartExecutableRequest { cell_name, executable } = request;
+    ) -> std::result::Result<Response<StartExecutableResponse>, Status> {
+        let ValidatedStartExecutableRequest { mut cell_name, executable } =
+            request;
 
         info!(
-            "CellService: start() cell_name={} executable={:?}",
+            "CellService: start() cell_name={:?} executable={:?}",
             cell_name, executable
         );
 
-        let mut cells = self.cells.lock().await;
-        let pid = cells.get_mut(&cell_name, move |cell| {
-            // TODO: `start_executable can potentially return a &Executable, and the we can
-            //    build `StartExecutalbeResponse` from it.
-            cell.start_executable(executable)
-        })?;
-
-        Ok(StartExecutableResponse { pid })
+        if cell_name.is_empty() {
+            // we are in the correct cell
+            let mut executables = self.executables.lock().await;
+            let executable = executables
+                .start(executable)
+                .map_err(CellsServiceError::ExecutablesError)?;
+
+            let pid = executable.pid().expect("pid").as_raw();
+            Ok(Response::new(StartExecutableResponse { pid }))
+        } else {
+            // we are in a parent cell
+            let child_cell_name = cell_name.pop_front().expect("len > 0");
+
+            let mut cells = self.cells.lock().await;
+            let client_config = cells
+                .get(&child_cell_name, move |cell| cell.client_config())
+                .map_err(CellsServiceError::CellsError)?;
+
+            // TODO: Handle error
+            let client = AuraeClient::new(client_config)
+                .await
+                .expect("failed to create AuraeClient");
+
+            // TODO: This seems wrong.
```

Comment:
i think it's ok if you imagine this on a separate service for managing executables.